packages/oauth/src/index.ts

```diff
@@ -71,6 +75,18 @@ export class InstallProvider {
     this.handleCallback = this.handleCallback.bind(this);
     this.authorize = this.authorize.bind(this);
     this.authVersion = authVersion;
+
+    this.authorizationUrl = authorizationUrl;
+    if (authorizationUrl !== 'https://slack.com/oauth/v2/authorize' && authVersion === 'v1') {
```

Comment:
This code is still new to me, but is there never a case where a user might (accidentally?) pass in `https://slack.com/oauth/authorize` and `v2`? If so, does that work without issue?

Reply:
Ooo good point! We don't do any conditional checks for `v2`, so in that scenario you describe, the `authorizationUrl` of `https://slack.com/oauth/authorize` will take priority and it should work as expected!

Reply:
i think this is a reasonable approach. the risk is that the developer sets the `authorizationUrl` to the `v1` value, so the response we get from the platform is in the v1 shape, and then when we parse it later it could throw or produce some unexpected result since `authVersion: 'v2'` and we expect it in that shape.

but i think if you're setting `authorizationUrl` on your own, you're asking to get down into the guts of how it all works. with great power come great responsibility, right?